File: csharp/DiagramPrinter/PrintableDiagram.cs
namespace DiagramPrinter;

public class PrintableDiagram
{

    public FlowchartDiagram Diagram { get; }
    
    public PrintableDiagram(FlowchartDiagram diagram)
    {
        Diagram = diagram;
    }

    public virtual DiagramMetadata GetDiagramMetadata()
    {
        return new DiagramMetadata(Diagram);
    }

    public virtual bool PrintToFile(string fromFilename, string targetFilename)
    {
        return Diagram.FlowchartAsPdf().CopyFile(fromFilename, targetFilename, true);
    }

    public virtual string SummaryInformation()
    {
        return Diagram.SummaryInformation();
    }
}

File: typescript/src/printableDiagram.ts
import {DiagramMetadata, FlowchartDiagram, IDiagramMetadata} from './documents';

export class PrintableDiagram {
  private readonly _diagram: FlowchartDiagram;

  constructor(diagram: FlowchartDiagram) {
    this._diagram = diagram;
  }

  getDiagram(): FlowchartDiagram {
    return this._diagram;
  }

  getDiagramMetadata(): IDiagramMetadata {
    return new DiagramMetadata(this._diagram);
  }

  printToFile(fromFilename: string, targetFilename: string): boolean {
    return this._diagram.flowchartAsPdf().copyFile(fromFilename, targetFilename, true);
  }

  printToSpreadsheetFile(fromFilename: string, targetFilename: string): boolean {
    return this._diagram.flowchartDataAsSpreadsheet().copyFile(fromFilename, targetFilename, true);
  }

  summaryInformation(): string {
    return this._diagram.summaryInformation();
  }
}
